frontend/src/screens/pairingGameScreen.tsx

```diff
@@ -34,10 +42,17 @@ export default function PairingGameScreen(props: PairingGameScreenProps) {
         if (currentRoundId == "") {
             fi.startRound(sessionId).then(result => setCurrentRoundId(result));
         } else {
-            setUser(user);
-            fi.getRoundPairs(currentRoundId).then(words => {
-                setEnglishWords(durstenfeldShuffle([words.map((word, i) => word.english)[0]]));
-                setTurkishWords(durstenfeldShuffle(words.map((word, i) => ({ turkish: word.turkish, correctEnglishWord: word.english }))));
+            fi.getUser().then(user => {
+                setUser(user);
+                fi.getRoundPairs(currentRoundId).then(words => {
+                    if (isPairingGame(user)) {
+                        setEnglishWords(durstenfeldShuffle(words.map((word, i) => word.english))); // pairing case
+                        setTurkishWords(durstenfeldShuffle(words.map((word, i) => ({ turkish: word.turkish, correctEnglishWord: word.english }))));
+                    } else {
+                        setEnglishWords(durstenfeldShuffle([words.map((word, i) => word.english)[0]])); // selecting case
+                        setTurkishWords(durstenfeldShuffle(words.map((word, i) => ({ turkish: word.turkish, correctEnglishWord: word.english }))));
+                    }
+                }).catch(console.error);
```

Comment:
I think setting the Turkish words is the same for both games, so we can pull that out of the conditional statement and just do setting the english words.